From: olh@suse.de
Subject: create an entry in modules.ofmap

mesh, mac53c94 and airport already have an entry.
Add the remaining drivers for pmac.
platinumfb uses also struct of_device_id, but it cant be a module.
ide may be a module in the future.

Signed-off-by: Olaf Hering <olh@suse.de>

 drivers/i2c/busses/i2c-keywest.c     |    1 +
 drivers/ide/ppc/pmac.c               |    1 +
 drivers/macintosh/mediabay.c         |    1 +
 drivers/macintosh/therm_pm72.c       |    1 +
 drivers/macintosh/therm_windtunnel.c |    1 +
 drivers/net/bmac.c                   |    1 +
 drivers/net/mace.c                   |    1 +
 sound/ppc/pmac.c                     |   32 ++++++++++++++++++++++++++++++++
 8 files changed, 39 insertions(+)

Index: linux-2.6.12/drivers/i2c/busses/i2c-keywest.c
===================================================================
--- linux-2.6.12.orig/drivers/i2c/busses/i2c-keywest.c
+++ linux-2.6.12/drivers/i2c/busses/i2c-keywest.c
@@ -729,6 +729,7 @@ static struct of_device_id i2c_keywest_m
 	},
 	{},
 };
+MODULE_DEVICE_TABLE (of, i2c_keywest_match);
 
 static struct macio_driver i2c_keywest_macio_driver = 
 {
Index: linux-2.6.12/drivers/ide/ppc/pmac.c
===================================================================
--- linux-2.6.12.orig/drivers/ide/ppc/pmac.c
+++ linux-2.6.12/drivers/ide/ppc/pmac.c
@@ -1653,6 +1653,7 @@ static struct of_device_id pmac_ide_maci
 	},
 	{},
 };
+MODULE_DEVICE_TABLE (of, pmac_ide_macio_match);
 
 static struct macio_driver pmac_ide_macio_driver = 
 {
Index: linux-2.6.12/drivers/macintosh/mediabay.c
===================================================================
--- linux-2.6.12.orig/drivers/macintosh/mediabay.c
+++ linux-2.6.12/drivers/macintosh/mediabay.c
@@ -816,6 +816,7 @@ static struct of_device_id media_bay_mat
 	},
 	{},
 };
+MODULE_DEVICE_TABLE (of, media_bay_match);
 
 static struct macio_driver media_bay_driver =
 {
Index: linux-2.6.12/drivers/macintosh/therm_pm72.c
===================================================================
--- linux-2.6.12.orig/drivers/macintosh/therm_pm72.c
+++ linux-2.6.12/drivers/macintosh/therm_pm72.c
@@ -2017,6 +2017,7 @@ static struct of_device_id fcu_match[] =
 	},
 	{},
 };
+MODULE_DEVICE_TABLE (of, fcu_match);
 
 static struct of_platform_driver fcu_of_platform_driver = 
 {
Index: linux-2.6.12/drivers/macintosh/therm_windtunnel.c
===================================================================
--- linux-2.6.12.orig/drivers/macintosh/therm_windtunnel.c
+++ linux-2.6.12/drivers/macintosh/therm_windtunnel.c
@@ -467,6 +467,7 @@ static struct of_device_id therm_of_matc
 	.compatible	= "adm1030"
     }, {}
 };
+MODULE_DEVICE_TABLE (of, therm_of_match);
 
 static struct of_platform_driver therm_of_driver = {
 	.name		= "temperature",
Index: linux-2.6.12/drivers/net/bmac.c
===================================================================
--- linux-2.6.12.orig/drivers/net/bmac.c
+++ linux-2.6.12/drivers/net/bmac.c
@@ -1658,6 +1658,7 @@ static struct of_device_id bmac_match[] 
 	},
 	{},
 };
+MODULE_DEVICE_TABLE (of, bmac_match);
 
 static struct macio_driver bmac_driver = 
 {
Index: linux-2.6.12/drivers/net/mace.c
===================================================================
--- linux-2.6.12.orig/drivers/net/mace.c
+++ linux-2.6.12/drivers/net/mace.c
@@ -1016,6 +1016,7 @@ static struct of_device_id mace_match[] 
 	},
 	{},
 };
+MODULE_DEVICE_TABLE (of, mace_match);
 
 static struct macio_driver mace_driver = 
 {
Index: linux-2.6.12/sound/ppc/pmac.c
===================================================================
--- linux-2.6.12.orig/sound/ppc/pmac.c
+++ linux-2.6.12/sound/ppc/pmac.c
@@ -871,6 +871,38 @@ static void __init detect_byte_swap(pmac
 		chip->can_duplex = 0;
 }
 
+static struct of_device_id alsa_match[] =
+{
+	{
+	.compatible	= "awacs"
+	},
+	{
+	.compatible	= "screamer"
+	},
+	{
+	.compatible	= "burgundy"
+	},
+	{
+	.compatible	= "daca"
+	},
+	{
+	.compatible	= "tumbler"
+	},
+	{
+	.compatible	= "snapper"
+	},
+	{
+	.compatible	= "AOAKeylargo"
+	},
+	{
+	.compatible	= "AOAbase"
+	},
+	{
+	.compatible	= "AOAK2"
+	},
+	{},
+};
+MODULE_DEVICE_TABLE (of, alsa_match);
 
 /*
  * detect a sound chip
